dtl-lexer/src/lorem.rs

```diff
@@ -0,0 +1,288 @@
+    "at",
+    "in",
+    "id",
+    "quos",
+    "reprehenderit",
+    "numquam",
+    "iste",
+    "fugiat",
+    "sit",
+    "inventore",
+    "beatae",
+    "repellendus",
+    "magnam",
+    "recusandae",
+    "quod",
+    "explicabo",
+    "doloremque",
+    "aperiam",
+    "consequatur",
+    "asperiores",
+    "commodi",
+    "option",
+    "dolor",
+    "labore",
+    "temporibus",
+    "repellat",
+    "veniam",
+    "architecto",
+    "est",
+    "esse",
+    "mollitia",
+    "nulla",
+    "a",
+    "similique",
+    "eos",
+    "alias",
+    "dolore",
+    "tenetur",
+    "deleniti",
+    "porro",
+    "facere",
+    "maxime",
+    "corrupti",
+];
+
+static COMMON_WORDS: [&str; 19] = [
+    "lorem",
+    "ipsum",
+    "dolor",
+    "sit",
+    "amet",
+    "consectetur",
+    "adipisicing",
+    "elit",
+    "sed",
+    "do",
+    "eiusmod",
+    "tempor",
+    "incididunt",
+    "ut",
+    "labore",
+    "et",
+    "dolore",
+    "magna",
+    "aliqua",
+];
+
+pub fn sentence() -> String {
+    use rand::Rng;
+    use rand::seq::SliceRandom;
+
+    let mut rng = rand::thread_rng();
+    let num_sections = rng.gen_range(1..=5);
+    let mut sections = Vec::with_capacity(num_sections);
+
+    for _ in 0..num_sections {
+        let num_words = rng.gen_range(3..=12);
+        let selected_words: Vec<&str> = WORDS
+            .choose_multiple(&mut rng, num_words)
+            .copied()
+            .collect();
+
+        sections.push(selected_words.join(" "));
+    }
+
+    let mut sentence = sections.join(", ");
+
+    if let Some(first) = sentence.chars().next() {
+        let upper = first.to_uppercase().to_string();
+        let rest = &sentence[first.len_utf8()..];
+        sentence = format!("{upper}{rest}");
+    }
+
+    let punctuation = if rng.gen_bool(0.5) { "?" } else { "." };
+    sentence.push_str(punctuation);
+
+    sentence
+}
+
+pub fn paragraph() -> String {
+    let num_sentences = rand::thread_rng().gen_range(1..=4);
+    (0..num_sentences)
+        .map(|_| sentence())
+        .collect::<Vec<String>>()
+        .join(" ")
+}
+
+pub fn paragraphs(count: usize, common: bool) -> Vec<String> {
+    let mut paras = Vec::with_capacity(count);
+
+    for i in 0..count {
+        if common && i == 0 {
+            paras.push(COMMON_P.to_string());
+        } else {
+            paras.push(paragraph());
+        }
+    }
+
+    paras
+}
+
+pub fn words(count: usize, common: bool) -> String {
+    if common && count <= COMMON_WORDS.len() {
+        return COMMON_WORDS[..count].join(" ");
+    }
+    let mut rng = rand::thread_rng();
+    let mut word_list: Vec<&str> = Vec::with_capacity(count);
+    if common {
+        word_list.extend(&COMMON_WORDS);
+    }
+    let mut remaining = count - word_list.len();
+    while remaining > 0 {
+        let take = remaining.min(WORDS.len());
+        let sampled = WORDS.choose_multiple(&mut rng, take);
+        word_list.extend(sampled.cloned());
+        remaining -= take;
```

Comment:
We could combine `remaining` and `count`:

```suggestion
pub fn words(mut count: usize, common: bool) -> String {
    if common && count <= COMMON_WORDS.len() {
        return COMMON_WORDS[..count].join(" ");
    }
    let mut rng = rand::thread_rng();
    let mut word_list: Vec<&str> = Vec::with_capacity(count);
    if common {
        word_list.extend(&COMMON_WORDS);
        count -= word_list.len();
    }
    while count > 0 {
        let take = count.min(WORDS.len());
        let sampled = WORDS.choose_multiple(&mut rng, take);
        word_list.extend(sampled.copied());
        count -= take;
```